### DISCUSSION TITLE
Assistance Needed with Bolt V1 Board ESC Configuration

### DISCUSSION THREAD

**Discussion by Doktordrone** [bitcraze/discussions#1230] [Q&A]
Hi Bitcraze forum,

Im sadly not a shady super coder and this is my first contact with the CrazyFly world :/ so I'm currently troubleshooting a Bolt V1.0 board, which has been flashed with the latest firmware. My objective is to configure it to run with
the Oneshot125 protocol on my 4in1 ESC. Despite prior testing to confirm
Oneshot compatibility of the ESC, I'm encountering difficulties in
establishing communication between the Bolt and the ESC.

Upon connecting to the client, the console displays the following messages:

running!
SYS: Build 0:e527895e8c6e (2023.11) CLEAN
SYS: I am 0x20373347344D500E004D0051 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 0 deck(s) found
IMU: BMI088: Using SPI interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushless motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.2 (CB10)
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 20496 bytes
STAB: Starting stabilizer loop
SUP: Can not fly

It appears that the Bolt utilizes the brushless motor driver, but the
specific protocol remains unclear. Could you please advise me on where I
can find help to adjust these settings in the cfclient and hopefully provide guidance on how to performing a motor
test via the console??
Alternatively, can I find pre built firmware with these settings activated to enable this device to take flight.

Thank you in advance for your assistance.

**Comment by tobbeanton** [comment]
The default protocol is the OneShot125 and to change it you need to [rebuild](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/) the firmware and select the different protocol using [menuconfig](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/kbuild/).

What ESC are you using and what beeps does it do during startup? Has the Bolt [calibrated](https://www.bitcraze.io/documentation/tutorials/getting-started-with-crazyflie-2-x/#leds) first, else it will not output any motor signal?

**Reply by Doktordrone** [reply]
Thanks! I'll experiment with Linux on my Windows this Crazyfly SW is intriguing! 😄 
It's a 4in1 ESC flashed with the latest BlHeli16 firmware, tested with Betaflight using Oneshot125. It beeps on startup as usual but motors don't spin after arming the quad cfclient and powering up via XBox360controller. I can see the software sending throttle signals to the ESC though...

**Reply by tobbeanton** [reply]
It sound a bit like there are no motor signals to the ESCs. Do you have a multimeter or Scope you could check with just to be sure your get something.

The Bolt outputs 3V signal but this "should" not be a problem, maybe it is for your ESC...? What brand and type is it?
